uaclient/entitlements/realtime.py

```diff
@@ -32,6 +32,13 @@ def _check_for_reboot(self) -> bool:
         event.needs_reboot(reboot_required)
         return reboot_required
 
+    def _get_variants(self) -> Dict[str, Type[UAEntitlement]]:
+        return {
+            GenericRealtime.variant_name: GenericRealtime,
+            NvidiaTegraRealtime.variant_name: NvidiaTegraRealtime,
+            IntelIotgRealtime.variant_name: IntelIotgRealtime,
+        }
```

Comment:
How do we indicate that the generic variant is the default?

`pro enable realtime-kernel` and `pro enable realtime-kernel --variant generic` should do the exact same thing. It looks like we'll get different entitlement objects for them though.

Reply:
Separately, I'd like the visibility of variants to be driven by the contract server. In two ways preferably (though maybe we can add one or both of these as follow up PRs)
1. variants defined in pro-client but not defined in the entitlement **should not** get displayed
2. variants defined in the entitlement but not defined in the pro-client **should** get displayed (with fallback generic descriptions)

I think (1) can be easily implemented in this function by not including variants in the return value unless they're found in the entitlement.

(2) is a bit trickier, but can probably be an anonymous subclass or something that is returned here.

Reply:
Part of my goal is that we should be able to merge and release this before the variants are finalized and released on the backend. Right now if you run this without overriding the machine_token then you get some weird behavior like all variants saying they're enabled at once.